src/util/modules.py

```diff
@@ -967,7 +967,9 @@ def new_module(cr, module, deps=(), auto_install=False, category=None, countries
     trigger_auto_install(cr, module)
 
 
-def _caller_version(depth=3):
+def _caller_version(depth=None):
+    if depth is None:
+        depth = 3 if version_gte("16.0") else 2
```

Comment:
Parsing the version is more costly than doing 1 more iteration
```suggestion
def _caller_version(depth=2):
```

Reply:
done. I thought it did on purpose.

Reply:
thank you for the fix!